client/verta/verta/pipeline/_pipeline_step.py

```diff
@@ -0,0 +1,213 @@
+# -*- coding: utf-8 -*-
+
+from typing import Any, Dict, List, Optional
+
+from verta._internal_utils._utils import Configuration, Connection
+from verta.registry.entities import RegisteredModel, RegisteredModelVersion
+
+
+class PipelineStep:
+    """
+    A single step within an inference pipeline, representing a single model version to be run.
+
+    Parameters
+    ----------
+    name : str
+        Name of the step, for use within the scope of the pipeline only.
+    model_version : :class:`~verta.registry.entities.RegisteredModelVersion`
+        Registered model version to run for the step.
+    predecessors : list, optional
+        List of PipelineSteps whose outputs will be treated as inputs to this step.
+        If not included, the step is assumed to be an initial step. Values must be unique.
+
+    Attributes
+    ----------
+        name : str
+            Name of the step within the scope of the pipeline.
+        model_version
+            :class:`~verta.registry.entities.RegisteredModelVersion` run by this step.
+        predecessors : list
+            List of PipelineSteps whose outputs will be treated as inputs to this step.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        model_version: RegisteredModelVersion,
+        predecessors: Optional[
+            List["PipelineStep"]
+        ] = None,  # Optional because it could be the first step with no predecessors
+    ):
+        self._name = name
+        self._model_version = model_version
+        self._predecessors = predecessors or list()
+
+        # avoid the need to pass in connection params when building as local object
+        self._registered_model: Optional[RegisteredModel] = None
+
+        # avoid additional http calls to refresh RMV cache.
+        self._registered_model_id = self._model_version.registered_model_id
+
+    def __repr__(self) -> str:
+        sequence = (
+            "\nPipelineStep:",
+            f"step name: {self.name}",
+        )
+        if self._registered_model:  # don't display on local-only objects`
+            sequence += (f"registered_model: {self._registered_model.name}",)
+        sequence += (
+            f"registered_model_version: {self.model_version.name}",
+            f"registered_model_version_id: {self.model_version.id}",
+            f"predecessors: {[s.name for s in self.predecessors]}",
+        )
+        return "\n".join(sequence)
```

Comment:
I'd recommend doing either a `"\n\t".join()` or `"\n    ".join()` like we do in other parts of the client

https://github.com/VertaAI/modeldb/blob/24b535d41e64b63a27705da729faa213a6c021a6/client/verta/verta/environment/_python.py#L147